hangman.py

```diff
@@ -0,0 +1,270 @@
+# Problem Set 2, hangman.py
+# Name: Bogdan Zadorozhny
+# Group: KM-02
+# Hangman Game
+# -----------------------------------
+import random
+import string
+
+
+WORDLIST_FILENAME = "words.txt"
+
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print("Loading word list from file...")
+    inFile = open(WORDLIST_FILENAME, 'r')
+    line = inFile.readline()
+    wordlist = line.split()
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+def choose_word(wordlist):
+    """
+    wordlist (list): list of words (strings)
+    Returns a word from wordlist at random
+    """
+    return random.choice(wordlist)
+
+
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: set (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    for letter in secret_word:
+        if letter not in letters_guessed:
+            return True
+    return False
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: set (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    compared_list = []
+    for i in secret_word:
+        if i in letters_guessed:
+            compared_list.append(i)
+        elif i not in letters_guessed:
+            compared_list.append("_ ")
+    compared_word = ""
```

Comment:
Actually - this empty string doesn't have anything common with comparing words. It's just a separator.